lib/pages/goal_page.dart

```diff
@@ -3,102 +3,485 @@ import 'package:linear/model/goal.dart';
+            TextButton(
+               onPressed: () {
+                filterMenu(2);
+                Navigator.pop(context);
+              },
+              child: const Text('Finished'),
+            ),
+            TextButton(
+               onPressed: () {
+                 filterMenu(3);
+                Navigator.pop(context);
+              },
+              child: const Text('All'),
+            ),
+          ];
+ }
+
+
+ List<Widget> sortChildren(context) {
+   return [
+            TextButton(
+              onPressed: () {
+                sortMenu(0);
+                Navigator.pop(context);
+              },
+              child: const Text('Newest'),
+            ),
+            TextButton(
+               onPressed: () {
+                sortMenu(1);
+                Navigator.pop(context);
+              },
+              child: const Text('Oldest'),
+            ),
+            TextButton(
+               onPressed: () {
+                sortMenu(2);
+                Navigator.pop(context);
+              },
+              child: const Text('Completion↑'),
+            ),
+            TextButton(
+               onPressed: () {
+                sortMenu(3);
+                Navigator.pop(context);
+              },
+              child: const Text('Completion↓'),
+            ),
+             TextButton(
+               onPressed: () {
+                sortMenu(4);
+                Navigator.pop(context);
+              },
+              child: const Text('Communities A to Z'),
+            ),
+             TextButton(
+               onPressed: () {
+                sortMenu(5);
+                Navigator.pop(context);
+              },
+              child: const Text('Communities Z to A'),
+            ),
+          ];
+ }
+ 
+ buildGoalsPageScreen(){
+   if (_isLoading) {
+     return buildGoalsPageLoadingScreen();
+   } else if (_isErrorFetchingGoals) {
+     return buildGoalsPageErrorScreen();
+   } else {
+     return buildGoalsPageContent();
+   }
+ }
+ 
+ buildGoalsPageLoadingScreen(){
+   return
+     Center(
+       child: const CircularProgressIndicator()
+     );
+ }
+ 
+ buildGoalsPageErrorScreen(){
+   return const Scaffold(
+       body: Center(
+         child: Text(
+           "We ran into an error trying to obtain the goals. \nPlease try again later.",
+           textAlign: TextAlign.center,
+         ),
+       ),
+     );
+ }
+ 
+ buildGoalsPageContent(){
+   return
+      Column(
+         crossAxisAlignment: CrossAxisAlignment.stretch,
+         children: [
+      const SizedBox(height: 10),
+      buildSearchBar(),
+      Row(children: [
+        Spacer(),
+        if(isSorting == true)
+        Text("Showing ${filteringBy} goals sorted by ${sortingBy}"),
+        Spacer(),
+      ]),
+       buildGoalsList(_goals),
+     ],
+    );
+ }
+ 
+  buildSearchBar(){
+    return Container(
+      margin: const EdgeInsets.all(10),
+      child: TextFormField(
+        controller: searchInput,
```

Comment:
side note: I believe all community names will be lowercase, if i search for the community 'soccer' by entering 'Soccer' will that still return 'soccer' if not converting the searchInput to lowercase or if there is a way to limit entry to only lower cases characters it may solve this issue.

Reply:
resolved, if you search with uppercase it will now convert to lowercase and it'll return